pkg/nsx/services/parse.go

```diff
@@ -0,0 +1,43 @@
+package services
+
+import (
+	"errors"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+func getRuleAction(rule *v1alpha1.SecurityPolicyRule) (string, error) {
+	ruleAction := util.ToUpper(*rule.Action)
+	validRuleActions := []string{
+		util.ToUpper(v1alpha1.RuleActionAllow),
+		util.ToUpper(v1alpha1.RuleActionDrop),
+		util.ToUpper(v1alpha1.RuleActionReject),
+	}
+	for _, validRuleAction := range validRuleActions {
+		if ruleAction == validRuleAction {
+			return ruleAction, nil
+		}
+	}
+	return "", errors.New("invalid rule action")
+}
+
+func getRuleDirection(rule *v1alpha1.SecurityPolicyRule) (string, error) {
+	ruleDirection := util.ToUpper(*rule.Direction)
+	if ruleDirection == util.ToUpper(v1alpha1.RuleDirectionIngress) ||
+		ruleDirection == util.ToUpper(v1alpha1.RuleDirectionIn) {
+		return "IN", nil
+	} else if ruleDirection == util.ToUpper(v1alpha1.RuleDirectionEgress) ||
```

Comment:
put util.ToUpper(v1alpha1.RuleDirectionEgress) util.ToUpper(v1alpha1.RuleDirectionOut) in a var, so don't transform them in each function call?

Reply:
ok